### PR TITLE
SWTCH-1189 display details of assets enrolment

### PR DESCRIPTION
**Pull request by dawidgil**

### CODE REVIEW

File: src/app/routes/assets/asset-enrolment-list/asset-enrolment-list.component.spec.ts [hunk 1]
```diff
@@ -1,25 +1,114 @@
...
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { ActivatedRoute, Params } from '@angular/router';
+import { BehaviorSubject } from 'rxjs';
+import { MockStore, provideMockStore } from '@ngrx/store/testing';
+import { UrlService } from '../../../shared/services/url-service/url.service';
+import { AssetDetailsActions, AssetDetailsSelectors } from '@state';
+import { RouterTestingModule } from '@angular/router/testing';
 
-xdescribe('AssetEnrolmentListComponent', () => {
+export class MockActivatedRoute {
```
Comment (jrhender) [inline src/app/routes/assets/asset-enrolment-list/asset-enrolment-list.component.spec.ts:12]: Is this class a generic class that could be shared by multiple tests?

File: src/app/shared/services/iam.service.ts [hunk 2]
```diff
@@ -69,6 +70,10 @@ export class IamService {
     return this._iam;
   }
 
+  getAssetById(id) {
+    return this.observableWrapper(this.iam.getAssetById({id}), {message: 'Getting selected asset data...'});
```
Comment (jrhender) [inline src/app/shared/services/iam.service.ts:74]: Is the reason for the `observableWrapper` to be able to put the `loadingService` `show()` and `hide()` around the `source`? Maybe a name like `wrapWithLoadingService` would be better?

File: src/app/shared/services/iam.service.ts [hunk 3]
```diff
@@ -167,4 +172,12 @@ export class IamService {
 
     return retVal;
   }
+
+  private observableWrapper<T>(source: Promise<T> | Observable<T>,
+                               loaderConfig?: { message?: string | string[]; cancelable?: boolean }) {
+    this.loadingService.show(loaderConfig.message, !!loaderConfig.cancelable);
```
Comment (jrhender) [inline src/app/shared/services/iam.service.ts:178]: If asserting with `!!` that `cancelable` is defined, wouldn't it better to require that in the param?
```suggestion
                               loaderConfig?: { message?: string | string[]; cancelable: boolean }) {
    this.loadingService.show(loaderConfig.message, loaderConfig.cancelable);
```

**Review by jrhender — APPROVED** [review]
A few comments but looks good!

File: src/app/routes/assets/asset-enrolment-list/asset-enrolment-list.component.spec.ts (hunk 1 above)
Comment (dawidgil) [reply src/app/routes/assets/asset-enrolment-list/asset-enrolment-list.component.spec.ts:12]: Yeah, I'm planning to do a directory with mocks/stubs/function supports for tests. This require some bigger refactoring in multiple places.

File: src/app/shared/services/iam.service.ts (hunk 3 above)
Comment (dawidgil) [reply src/app/shared/services/iam.service.ts:178]: No, because not always it is required and not always defined.

**Pull request merged**